earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -285,90 +284,79 @@ void PacketGenAssetVersionImplD::UpdateChildren(
-          levInputVers.push_back(inset.combinedrp_);
-          levInputs.push_back(blendVer.Ref());
-          levInputVers.push_back(blendVer);
 
           // Note: Before creating a merge inset, we check if the cached alpha
           // blend directory exists for the asset since in GEE-5.x we need to
           // pick up the GEE-4.x Imagery/Terrain Projects (the PacketLevel
           // assets) that have no alpha blend cached.
           const std::string cache_alpha_path =
-              blendVer->GetOutputFilename(0) + "/cache_alpha.pack";
-          const bool cache_alpha_exist = khExists(cache_alpha_path);
-
-          packetLevelConfig.insets.push_back
-            (PacketLevelConfig::Inset(
+              levInputVers[i+1]->GetOutputFilename(0) + "/cache_alpha.pack";
+          if (khExists(cache_alpha_path)) {
+            cached_blend_alpha = levInputs[i+1];
+          }
+          cached_blend = levInputs[i+1];
+        }
+        else {
+          emptyElements.push_back(i+1);
+        }
+        packetLevelConfig.insets[idx] =
+            PacketLevelConfig::Inset(
                 inset.dataRPFile,
                 inset.alphaRPFile,
-                blendVer.Ref(), /* cached blend */
-                cache_alpha_exist ?
-                    blendVer.Ref() : SharedString() /* cached blend alpha */));
-        } else {
-          // This is a subsequent merge (from cached merges) It's
-          // possible that earlier merges also short circuited the
-          // merge with opaque top tiles. So we may need to pull
-          // from the product too.
-
-          // NOTE: For now this case should be identical to the
-          // one above. Maybe later we can put some more smarts
-          // here to limit dependencies on the products?
-
-          levInputs.push_back(inset.combinedrp_->GetRef());
-          levInputVers.push_back(inset.combinedrp_);
-          levInputs.push_back(blendVer.Ref());
-          levInputVers.push_back(blendVer);
-
-          // Note: Before creating a merge inset, we check if the cached alpha
-          // blend directory exists for the asset since in GEE-5.x we need to
-          // pick up the GEE-4.x Imagery/Terrain Projects (the PacketLevel
-          // assets) that have no alpha blend cached.
-          const std::string cache_alpha_path =
-              blendVer->GetOutputFilename(0) + "/cache_alpha.pack";
-          const bool cache_alpha_exist = khExists(cache_alpha_path);
+                cached_blend,
+                cached_blend_alpha);
+      }
 
-          packetLevelConfig.insets.push_back
-            (PacketLevelConfig::Inset(
-                inset.dataRPFile,
-                inset.alphaRPFile,
-                blendVer.Ref(), /* cached blend */
-                cache_alpha_exist ?
-                    blendVer.Ref() : SharedString()  /* cached blend alpha */));
-        }
+      for (size_t idx = 0; idx < emptyElements.size(); ++idx) {
```

Comment:
I think this loop needs to be iterating emptyElements in reverse. Otherwise the wrong elements will be getting erased from `levInputs` and `levInputVers` as elements get shifted after each `erase` call.